storage/sqlite/sqlite.go

```diff
@@ -51,6 +51,49 @@ func (r *Backend) Open(options url.Values) {
 	r.tenant = make(map[string]*sql.DB)
 }
 
+func (r Backend) GetLastDataItems(tenant string, id string, numOfItems int) ([]storage.DataItem, error) {
+	res := make([]storage.DataItem, numOfItems)
+	db, _ := r.GetTenant(tenant)
+	var count int
+	// check if id exist
+	if !r.IdExist(tenant, id) {
+		return nil, fmt.Errorf("Metric does not exist in database.")
+	}
+
+	countQuery := fmt.Sprintf("select count(timestamp) from %s", id)
+	db.QueryRow(countQuery).Scan(&count)
+
+	if count < numOfItems {
```

Comment:
No need to handle the error here, let the caller handle it, much simpler and easy to test 😄

You have this check is in all the plugins

p.s.
a. currently we only need one last item.
b. if we want 10 items and we got 5, this 5 items are better then nil.

Reply:
Cool :+1: